src/batch.rs

```diff
@@ -453,3 +460,30 @@ impl RasterBatch {
         false
     }
 }
+
+trait InsideTest<T> {
+    fn contains_may_be_false_negative(&self, clip: &T) -> bool;
```

Comment:
I would just call this `might_contain` and clarify in the comments that this may be a false negative.